processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go

```diff
@@ -91,6 +117,59 @@ func (tsp *tracesamplerprocessor) Shutdown() error {
 	return nil
 }
 
+// parseSpanSamplingPriority checks if the span has the "sampling.priority" tag to
+// decide if the span should be sampled or not. The usage of the tag follows the
+// OpenTracing semantic tags:
+// https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table
+func parseSpanSamplingPriority(span *tracepb.Span) samplingPriority {
+	attribMap := span.GetAttributes().GetAttributeMap()
+	if attribMap == nil {
+		return deferDecision
+	}
+
+	samplingPriorityAttrib := attribMap["sampling.priority"]
+	if samplingPriorityAttrib == nil {
+		return deferDecision
+	}
+
+	decideForDoubleFn := func(value float64) samplingPriority {
+		if value == 0.0 {
+			return doNotSampleSpan
+		} else if value > 0.0 {
+			return mustSampleSpan
+		}
+		return deferDecision
+	}
+
+	// By default defer the decision.
+	decision := deferDecision
+
+	// Try check for different types since there are various client libraries
+	// using different conventions regarding "sampling.priority". Besides the
+	// client libraries it is also possible that the type was lost in translation
+	// between different formats.
+	switch samplingPriorityAttrib.Value.(type) {
+	case *tracepb.AttributeValue_IntValue:
+		value := samplingPriorityAttrib.GetIntValue()
+		if value == 0 {
+			decision = doNotSampleSpan
+		} else if value > 0 {
+			decision = mustSampleSpan
+		}
+	case *tracepb.AttributeValue_DoubleValue:
+		value := samplingPriorityAttrib.GetDoubleValue()
+		decision = decideForDoubleFn(value)
+	case *tracepb.AttributeValue_StringValue:
+		if attribVal := samplingPriorityAttrib.GetStringValue().GetValue(); attribVal != "" {
+			if value, err := strconv.ParseFloat(attribVal, 64); err == nil {
```

Comment:
Checking for the empty string is just an optimization - and I'm not sure it is worthwhile - `strconv.ParseFloat("", ...)` will return `err != nil`.